src/batch/src/task/task_execution.rs

```diff
@@ -333,6 +337,11 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
                             }
                         }
                     }
+
+                    if should_stop {
+                        info!("Task done!");
+                        break;
+                    }
```

Comment:
Actually I think we should break above - as soon as we receive a None.

And send the None outside the loop. Same for above when we receive from shutdown channel

Reply:
Good point.

Reply:
BTW we can use `?` on the `.send` error now.